### PR TITLE
Timer in runtime

### PR DESCRIPTION
**Pull request by zappen999**
### Description

Implements a timer to use for monitoring of time-critical tasks
### Example usage

``` js
// Setup reporting. In this case to a log
Timing.report(function handleTimingReport(level, name, actual, assertions) {
  const message = `Profiler: Action '${name}' took ${actual}ms to complete. ` +
    `Expected to be below ${assertions.info}ms`;

  if (level === Timing.levels.info) {
    return logger.info(message, assertions);
  }

  return logger.warn(message, assertions);
});

// Expect this task to be done within 20ms, with a limit of 100 (raises warning)
const asyncTimer = new Timing.Check('Some async task', 20, 100);

doSomething().then(function() {
  asyncTimer.stop();
});
```
### Related issues
#5

### CODE REVIEW

File: src/lib/timing/index.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  // Always report if debug level
+  if (settings.currentLevel === levels.debug) {
+    return settings.reportHandler(levels.debug, this.action, diff, assertions);
+  }
+
+  if (diff < this.infoThreshold) {
+    return;
+  }
+
+  let reportLevel = diff < this.warnThreshold ? levels.info : levels.warn;
```
Comment (JoelRoxell) [inline src/lib/timing/index.js]: Should be const, missing line break before return statement.

File: src/lib/timing/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+const expect = chai.expect;
+const sinon = require('sinon');
+
+const Timing = require('./index');
+
+describe('Profiling', function() {
+  describe('Report', function() {
+    it('should set callback function', function() {
+      const someCallback = function() {};
+      Timing.report(someCallback);
```
Comment (JoelRoxell) [inline src/lib/timing/index.spec.js:12]: Assignment operations and function calls should be grouped. (New line)

**Review by JoelRoxell — CHANGES_REQUESTED** [review]

File: src/lib/timing/index.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  return (hrResult[0] * 1000) + (hrResult[1] / 1e6);
+}
+
+/**
+ * Creates a new check
+ * @param {String}  action         Name of the action to measure
+ * @param {Integer} infoThreshold  Millisecond threshold to emit info
+ * @param {Integer} warnThreshold  Millisecond threshold to emit warning
+ */
+function Check(action, infoThreshold, warnThreshold) {
```
Comment (LinusLjung) [inline src/lib/timing/index.js:55]: Are you sure your example will work? This method doesn't return anything

**Review by JoelRoxell — APPROVED** [review]

File: src/lib/timing/index.js (hunk 3 above)
Comment (zappen999) [reply src/lib/timing/index.js:55]: Yes. It will work, since i use the `new` keyword which will create an object and return it.

**Pull request merged**